File: src/main/java/com/skyspacefiftyfive/service/CheckinDetailService.java
package com.skyspacefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefiftyfive.domain.CheckinDetail;
import com.skyspacefiftyfive.dto.CheckinDetailDTO;
import com.skyspacefiftyfive.dto.CheckinDetailSearchDTO;
import com.skyspacefiftyfive.dto.CheckinDetailPageDTO;
import com.skyspacefiftyfive.dto.CheckinDetailConvertCriteriaDTO;
import com.skyspacefiftyfive.service.GenericService;
import com.skyspacefiftyfive.dto.common.RequestDTO;
import com.skyspacefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CheckinDetailService extends GenericService<CheckinDetail, Integer> {

	List<CheckinDetail> findAll();

	ResultDTO addCheckinDetail(CheckinDetailDTO checkinDetailDTO, RequestDTO requestDTO);

	ResultDTO updateCheckinDetail(CheckinDetailDTO checkinDetailDTO, RequestDTO requestDTO);

    Page<CheckinDetail> getAllCheckinDetails(Pageable pageable);

    Page<CheckinDetail> getAllCheckinDetails(Specification<CheckinDetail> spec, Pageable pageable);

	ResponseEntity<CheckinDetailPageDTO> getCheckinDetails(CheckinDetailSearchDTO checkinDetailSearchDTO);
	
	List<CheckinDetailDTO> convertCheckinDetailsToCheckinDetailDTOs(List<CheckinDetail> checkinDetails, CheckinDetailConvertCriteriaDTO convertCriteria);

	CheckinDetailDTO getCheckinDetailDTOById(Integer checkinDetailId);







}

File: skyspacefiftyfive_ui/src/services/CheckinDetailService.js
import http from "../http-common"; 

class CheckinDetailService {
  getAllCheckinDetails(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/checkinDetail/checkinDetails`, searchDTO);
  }

  get(checkinDetailId) {
    return this.getRequest(`/checkinDetail/${checkinDetailId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/checkinDetail?field=${matchData}`, null);
  }

  addCheckinDetail(data) {
    return http.post("/checkinDetail/addCheckinDetail", data);
  }

  update(data) {
  	return http.post("/checkinDetail/updateCheckinDetail", data);
  }
  
  uploadImage(data,checkinDetailId) {
  	return http.postForm("/checkinDetail/uploadImage/"+checkinDetailId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CheckinDetailService();
